pkg/operator/apiserver/controller/apiservice/apiservice_controller.go

```diff
@@ -143,20 +154,40 @@ func (c *APIServiceController) sync(ctx context.Context, syncCtx factory.SyncCon
 		cond.Reason = "Error"
 		cond.Message = err.Error()
 	}
-	if _, _, updateError := v1helpers.UpdateStatus(ctx, c.operatorClient, v1helpers.UpdateConditionFn(cond)); updateError != nil {
+	if _, _, updateError := v1helpers.UpdateStatus(ctx, c.operatorClient, v1helpers.UpdateConditionFn(cond), v1helpers.UpdateConditionFn(conditionAPIServicesDegraded)); updateError != nil {
 		if err == nil {
-			return updateError
+			return errors.NewAggregate([]error{updateError, syncDisabledAPIServicesErr})
 		}
 	}
 
-	return err
+	return errors.NewAggregate([]error{err, syncDisabledAPIServicesErr})
 }
 
-func (c *APIServiceController) syncAPIServices(ctx context.Context, apiServices []*apiregistrationv1.APIService, recorder events.Recorder) error {
+func (c *APIServiceController) syncDisabledAPIServices(ctx context.Context, apiServices []*apiregistrationv1.APIService) error {
 	errs := []error{}
-	var availableConditionMessages []string
 
 	for _, apiService := range apiServices {
+		if apiServiceObj, err := c.apiservicelister.Get(apiService.Name); err == nil {
+			if apiServiceObj.DeletionTimestamp != nil {
+				continue
```

Comment:
Technically objects can stuck in deletion (finalisers) for a long time. 
Do we want to handle this case somehow (log, set something on the condition) ?

Reply:
`syncEnabledAPIServices` returns an error when the corresponding api service Available condition is not ready. Updated the code to return error when a corresponding service is not deleted yet.

Reply:
update: we decided to log instead of returning the error